### ISSUE TITLE
VSIX install failed

### ISSUE THREAD

**Issue by Licantrop0** [Microsoft/WindowsTemplateStudio#1532]
**<WTS_team>**

**What happened**
Currently the belief is, on updating, the prior version fails to uninstall for some reason, (maybe a file lock) but the new version still installs and updates the VS configuration.  This gets VS extension manager into a state where it sees both but can't uninstall the older version.

**First item we need in helping to debug are logs:**
We’ll need the full setup logs by running https://aka.ms/vscollect.exe (please place the zip here in the issue). %temp% folder and copy VSLogs.zip.  If that fails, please email me them at crutkas@microsoft.com  We need the install logs to help see why we are getting into this weird state.

**Manual workaround:**  
1. Look at logs and validate folder.  should be in: C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2017\ENTERPRISE\COMMON7\IDE\EXTENSIONS\ with a folder something in tune of: _INU2PILL.SXO_
2. delete folder
3. Open VS command line prompt in admin mode
4. Run command: devenv /updateconfiguration
5. VS will now see the newer version / be able to update.

**</WTS_team>**

Tried to update to latest version, got these screens:
![image](https://user-images.githubusercontent.com/2388430/33698091-2a692348-dabf-11e7-99c0-e2147d041d9d.png)
![image](https://user-images.githubusercontent.com/2388430/33698127-56533570-dabf-11e7-99f8-5055e7183d24.png)

logs:
[VSIXInstaller_c55b6715-c73c-434c-8205-b369bf257f49.log](https://github.com/Microsoft/WindowsTemplateStudio/files/1537738/VSIXInstaller_c55b6715-c73c-434c-8205-b369bf257f49.log)

**Comment by sibille** [comment]
It seems that the VSIX installer did not fully uninstall the previous WTS Version (1.4). 
Try to uninstall WTS. 
If it fails, try to manually delete the content from C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2017\ENTERPRISE\COMMON7\IDE\EXTENSIONS\INU2PILL.SXO
Then try to install WTS 1.5 again from the gallery, the marketplace or using VS. 

If you encounter any issues please let us know.

**Comment by crutkas** [comment]
@sibille can we make the vsix installer more robust in any way?

**Comment by sibille** [comment]
The vsix installation process is managed by a Visual Studio, I think there is nothing we can do here.

**Comment by graymatter00** [comment]
Same problem here, using VS 2017 Community.

Tried uninstalling via the Extensions and Updates window.  This failed with same error message as the attempted install of the new version.

Had to delete the INU2PILL.SXO folder, then uninstall old version via the Extensions and Updates window, which now succeeded.  Then installed new version via the Extensions and Updates window.

Seems OK now.

**Comment by ralarcon** [comment]
Thanks for your feedback @graymatter00, looks like sometimes the VsixInstaller didn't completely remove the old version and need to finish manually. Wonder to understand if the root cause is that any resource is in use while trying to unisntall, but it should not as the uninstall process is launched once the Visual Studio instance is closed...

**Comment by v-tbert** [comment]
I just hit this too and followed same process as @graymatter00  to resolve it. FWIW I had this same problem with Powershell Pro extension a few months ago. It does sound like VS servicing of VSIX installs needs to be made more robust.

**Comment by crutkas** [comment]
We're proactively looking into this with Visual Studio.

**Comment by Licantrop0** [comment]
@sibille also uninstalling failed: 
[VSIXInstaller_25bde7d9-a8ea-488e-a5e1-4e344438cb26.log](https://github.com/Microsoft/WindowsTemplateStudio/files/1564555/VSIXInstaller_25bde7d9-a8ea-488e-a5e1-4e344438cb26.log)
Had to manually remove the folder, uninstall, and reinstall again like you suggested.

**Comment by crutkas** [comment]
@Licantrop0 @graymatter00 and @v-tbert.

We’ll need the full setup logs by running https://aka.ms/vscollect.exe (please place the zip here in the issue). 

VS team has two theories we're trying to validate.

**Comment by crutkas** [comment]
if you guys feel that you don't want to publicly share them, you can put them on onedrive / dropbox and directly email me a link, crutkas@microsoft.com

**Comment by v-tbert** [comment]
Will that produce valuable output if we already applied the workaround?

**Comment by crutkas** [comment]
@v-tbert i believe it will as it is grabbing logs.  I'll mention that to the VS team that you did a manual workaround when looking at your logs

**Comment by crutkas** [comment]
this is odd, i used to hit issues uninstalling / upgrading on one of my boxes but all "just worked" now

**Comment by crutkas** [comment]
got a capture!  Also updated the question of where the logs actually are, %temp% folder and copy VSLogs.zip

**Comment by crutkas** [comment]
Worked with the Visual Studio extension team this morning.  We know what happened but we don't know why.  My log capture was missing the install log on what actually got us into a double install state.

**What happened**
Currently the belief is, on updating, the prior version fails to uninstall for some reason, (maybe a file lock) but the new version still installs and updates the VS configuration.  This gets VS extension manager into a state where it sees both but can't uninstall the older version.

**First item we need in helping to debug are logs:**
We’ll need the full setup logs by running https://aka.ms/vscollect.exe (please place the zip here in the issue). %temp% folder and copy VSLogs.zip.  If that fails, please email me them at crutkas@microsoft.com  We need the install logs to help see why we are getting into this weird state.

**Manual workaround:**  
1. Look at logs and validate folder.  should be in: C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2017\ENTERPRISE\COMMON7\IDE\EXTENSIONS\ with a folder something in tune of: _INU2PILL.SXO_
2. Open VS command line prompt in admin mode
3. Run command: devenv /updateconfiguration
4. VS will now see the newer version / be able to update.

**Comment by LanceMcCarthy** [comment]
I attempted the manual workaround and it failed with the same error. 

Here are the steps I took:

1 - Confirmed the `inu2pill.sxo` folder exists (it's empty with the exception of a cache.bin file)
2,3 - Ran update configuration from elevated VS2017 developer CLI
4 - Downloaded a fresh copy of the VSIX from the VS Marketplace [here](https://marketplace.visualstudio.com/items?itemName=WASTeamAccount.WindowsTemplateStudio)
5 - Ran the downloaded VSIX

The same error occurs.

**Comment by crutkas** [comment]
i think i forgot the critical "delete folder"

**Comment by xasx** [comment]
Interestingly though, when following the Deletion method, i.e. 
1. delete the sub-dir under Extensions
2. uninstall the extension from VS

I see that the installation log pretends to have the newer version uninstalled, i.e. the one that could not be upgraded to.

Fail to upgrade:
[noupgrade-VSIXInstaller_8ac10232-e03b-40ca-b1d3-4b64f14ecc3a.log](https://github.com/Microsoft/WindowsTemplateStudio/files/1594795/noupgrade-VSIXInstaller_8ac10232-e03b-40ca-b1d3-4b64f14ecc3a.log)

Successfully uninstall after delete:
[uninstall-after-delete-VSIXInstaller_1254de33-b513-4aad-b9dd-fdcf5f3e96a7.log](https://github.com/Microsoft/WindowsTemplateStudio/files/1594794/uninstall-after-delete-VSIXInstaller_1254de33-b513-4aad-b9dd-fdcf5f3e96a7.log)

**Comment by Morhpeuss** [comment]
Hi, I have the same problem with the extension. But I didn't find the folder "inu2pill.sxo" under Extensions so I can't delete it.

**Comment by ralarcon** [comment]
@Morhpeuss That is because the folder inu2pill.sxo is for other version different from what you have blocked... in the log you should be able to find the exact folder for your version within C:\PROGRAM FILES (X86)\MICROSOFT VISUAL STUDIO\2017\ENTERPRISE\COMMON7\IDE\EXTENSIONS\. Other option is to find the sub-folder which have, for example, the file "Microsoft.Templates.2017.dll"

**Comment by Morhpeuss** [comment]
Thanks, I have found 2 folders with this dll. Should I delete both?

**Comment by ralarcon** [comment]
Yes, delete both. The new installation (as is a higher version) will leave the folders as needed.

**Comment by crutkas** [comment]
@Morhpeuss any chance you can hook us up with logs?

**Comment by crutkas** [comment]
@xasx thank you for providing logs, they have been passed to the VS team

**Comment by gordonwatts** [comment]
I have this exact problem. I've captured logs and sent them to you in an email (let me know if they don't get through). Removing the folder as instructed above seems to have fixed it.

**Comment by DominikErnst** [comment]
1) Removing 2 folders "2x24zybx.cdw" and "zfzeussl.kda" does not help on my machine.
C:\Program Files (x86)\Microsoft Visual Studio\2017\**Professional**\Common7\IDE\Extensions
2) Updated Visual Studio 2017 to 15.5.3.
3) Reinstalled WTS from [here](https://marketplace.visualstudio.com/items?itemName=WASTeamAccount.WindowsTemplateStudio)

Now it WORKS!

**Comment by ralarcon** [comment]
Moving this to next version as it is still under investigation.

**Comment by chenjiawei-19881015** [comment]
I have this exact problem. I've captured logs and sent them to you in an email (let me know if they don't get through). Removing the folder as instructed above seems to have fixed it.

**Comment by sibille** [comment]
Moving to backlog as still under investigation. @crutkas, did we get more info on this?

**Comment by LanceMcCarthy** [comment]
I can confirm this workaround worked on a machine that was having tons of problems.

**Comment by crutkas** [comment]
@LanceMcCarthy did you have nightlies and what not installed on top of RTM?

**Comment by crutkas** [comment]
@sibille has there bee any forward movement on this?

**Comment by sibille** [comment]
I don't think so, VS Team was working on this,

**Comment by crutkas** [comment]
can you continue to push on them to get resolution?

**Comment by sibille** [comment]
sure

**Comment by LanceMcCarthy** [comment]
I'm not sure how this was eventually fixed for me. Ultimately I found many copies of the extension's assembly (the extension had multiple IDs, probably from early or nightly build versions), I deleted them all manually and then ran updateconfig. 

Initially the template wasn't listed after updateconfig (as seen in my last reply), but now it is. Maybe it appeared after updating VS because there were a few updates since I had the issue, however I can't confirm if that is actually what fixed it.
